consistenthash/consistenthash.go

```diff
@@ -19,24 +19,81 @@ package consistenthash
 
 import (
 	"hash/crc32"
+	"math/bits"
 	"sort"
 	"strconv"
 )
 
 type Hash func(data []byte) uint32
 
+const defaultHashExpansion = 6
+
 type Map struct {
-	hash     Hash
-	replicas int
-	keys     []int // Sorted
-	hashMap  map[int]string
+	// Inputs
+
+	// hash is the hash function that will be applied to both added
+	// keys and fetched keys
+	hash                 Hash
+
+	// replicas is the number of virtual nodes that will be inserted
+	// into the consistent hash ring for each key added
+	replicas             int
+
+	// prefixTableExpansion is the multiple of virtual nodes that
+	// will be inserted into the internal hash table for O(1) lookups.
+	prefixTableExpansion int
+
+
+
+	// Internal data
+
+	// keys is the hash of the virtual nodes, sorted by hash value
+	keys    []int // Sorted
+
+	// hashMap maps the hashed keys back to the input strings.
+	// Note that all virtual nodes will map back to the same input
+	// string
+	hashMap map[int]string
+
+	// prefixShift is the number of bits an input hash should
+	// be right-shifted to act as a lookup in the prefixTable
+	prefixShift uint32
+
+	// prefixTable is a map of the most significant bits of
+	// a hash value to output all hashes with that prefix
+	// map to.  If the result is ambiguous (i.e. there is a
+	// hash range split within this prefix) the value will
+	// be blank and we should fall back to a binary search
+	// through keys to find the exact output
+	prefixTable []string
 }
 
+// New returns a blank consistent hash ring that will return
+// the key whos hash comes next after the hash of the input to
+// Get().
+// Increasing the number of replicas will improve the smoothness
+// of the hash ring and reduce the data moved when adding/removing
+// nodes, at the cost of more memory.
 func New(replicas int, fn Hash) *Map {
+	return NewConsistentHash(replicas, defaultHashExpansion, fn)
+}
+
+
+// NewConsistentHash returns a blank consistent hash ring that will return
+// the key whos hash comes next after the hash of the input to
+// Get().
```

Comment:
Go docs don't conventionally include the parens when talking about methods.

But you can spell it out more to be clear: "to the Map.Get method", etc.